src/cbsodatav4_kcvanderlinden/cbsodatav4.py

```diff
@@ -2,142 +2,188 @@
 import requests
 from requests.adapters import HTTPAdapter
 from urllib3.util import Retry
-import concurrent.futures
 import os
-import math
+import sys
+import concurrent.futures
+import json
+import dask.dataframe as dd
+import dask
+import random
+from time import sleep
 
 basic_odata_url = 'https://datasets.cbs.nl/odata/v1/CBS/'
 
+def DataFrame(tableID:str, name:str=None, limit:int=None, dataFilter:str=None, save_csv_path:str=None):
+    """
+    Return a Pandas DataFrame containing the data from the CBS OData API.
+
+    Parameters
+    ----------
+    tableID : str
+        The ID of the table to retrieve data from.
+    name : str, optional
+        The name of the specific table to retrieve, by default None.
+        If not specified, the entire dataset will be retrieved.
+    limit : int, optional
+        The maximum number of rows to retrieve, by default None.
+        If not specified, all rows will be retrieved.
+    dataFilter : str, optional
+        A filter to apply to the data, by default None.
+        If not specified, no filter will be applied.
+    save_csv_path : str, optional
+        if not None, save the csv file to the specified directory. Path must incude the filename.
+
+    Returns
+    -------
+    Pandas DataFrame
+        A DataFrame containing the retrieved data.
+    """
+    
+    if name == None:
+        df = fullDataset(tableID, limit, dataFilter)
+    else:
+        df = specificTable(tableID, name, limit, dataFilter)
+    # if cache is true, check if cache folder exists, otherwise create it
+    if save_csv_path is not None:
+        df.to_csv(f"{save_csv_path}.csv", index=False, single_file=True)
+        sys.stdout.write(f'\nwritten table to csv')
+    df = df.compute()
+    delete_json_files()
```

Comment:
Kan het gebeuren dat de json bestanden wel aangemaakt worden, maar er daarna een error ontstaat? Dan zou wellicht deze regel in een finally moeten.